site/source/docs/porting/Audio.rst

```diff
@@ -0,0 +1,143 @@
+
+Of course, the current OpenAL implementation strives to be memory-efficient, but it has no control over lower-level mechanisms involved.
+
+
+.. _Audio-openal-capture-behavior-g:
+
+Emscripten-specific capture behavior
+====================================
+
+Attempting to open an input stream to the user's audio capture device
+results in the asynchronous appearance of a small browser-specific dialog
+asking for the user's permission.
+
+.. note:: 
+    On Firefox, the user is also given the opportunity to pick from a list of devices at that moment.
+    On other browsers, this is normally a simple "Allow/Deny" button pair on a toast.
+
+With this in mind, when ``alcCaptureOpenDevice()`` is called with valid and 
+supported parameters, a "proxy" device is returned, which successfully 
+captures no sample until the user clicks "Allow" on said dialog.
+
+That means, when calling ``alcGetIntegerv(device, ALC_CAPTURE_SAMPLES, 1, &nframes)``, ``nframes`` remains set to zero until the user clicks "Allow". You might want to make your app handle this specific behavior.
+
+If the user clicks "Deny", the device is invalidated (because this is somewhat
+similar to unplugging the physical device) and calls to ``alcCapture*`` functions on that
+device then consistently fail with ``ALC_INVALID_DEVICE``.
+Your application must be prepared to handle this properly.
+
+.. note::
+    Some browsers "remember" this choice and apply it automatically every time it would be asked again instead.
+    There's no way for the implementation to detect this behavior.
+
+
+.. _Audio-openal-capture-details-g:
+
+Useful implementation details of OpenAL capture
+===============================================
+
+Internally, Web Audio's capture data is always backed by a Javascript ``Float32Array``.  
+Thus, ``AL_FORMAT_MONO_FLOAT32`` and ``AL_FORMAT_STEREO_FLOAT32`` are the only formats which do not require running "type conversion passes" on acquired buffers.
+
+Note, however, that :
+
+- It's probably pointless to pick a stereo format for capture;
+- float32 samples use more memory than the standard 16-bit or 8-bit integer formats;
+- Most microphones aren't very high quality.
+  Especially for real-time apps such as games, users will likely tolerate lower quality recordings.
+
+Also, the actual sample rate at which samples are acquired from the device is unpredictable and depends on many factors, including the user's browser and setup. If this sample rate does not match the one your app requests, the implementation is required to perform resampling on your behalf.  
+
+That sample rate is given by ``audioCtx.sampleRate``, where ``audioCtx`` is the ``AudioContext`` object used internally by the relevant capture ``ALCdevice``.
+Currently, Emscripten provides no direct way for applications to access this value, but this might be provided through an Emscripten-specific OpenAL extension (which is not here yet because it requires registration).
```

Comment:
Does this reference to asking for the actual sampling rate of an already created context, or the default sampling rate of created contexts? Adding functions for either as needed sounds good.